pom.xml

```diff
@@ -482,10 +483,22 @@
                                         <exclude>log4j:log4j</exclude>
                                     </excludes>
                                 </bannedDependencies>
+                                <!-- Ensure no dependencies are using incompatible versions of java -->
+                                <enforceBytecodeVersion>
+                                    <maxJdkVersion>${java.version}</maxJdkVersion>
+                                    <ignoredScopes>test</ignoredScopes>
```

Comment:
I don't know how frequent [such a usecase](https://docs.oracle.com/en/java/javase/17/docs/specs/jar/jar.html#multi-release-jar-files) is but i'm tempted to put `<strict>true</strict>` as well here. By default it's set to [false](https://github.com/mojohaus/extra-enforcer-rules/blob/master/src/main/java/org/codehaus/mojo/extraenforcer/dependencies/EnforceBytecodeVersion.java#L190) and could lead to issues? Thoughts?

Reply:
I'll try it out. It is better to be as "strict" as possible for now :)

Reply:
This causing lots of breakages: jackson, lombok, log4j all trigger the enforcer rule now